app/services/data_centers/merge_duplicates.rb

```diff
@@ -2,203 +2,59 @@ module DataCenters
   # WARNING: Run service with "run_update: true" argument and you'll update, 
   # without you'll get only logs and see what will be updated.
   class MergeDuplicates
-    LOG_PATH = Rails.root.join("log", "#{self.name.demodulize.underscore}.log")
-
-    def initialize(run_update: false)
-      @logger ||= Logger.new(LOG_PATH)
-      @run_update = run_update
+    def initialize(switch_all: true)
+      @switch_all = switch_all
+      @validator_ips_to_detach = []
     end
 
     def call
-      log_message("Run update: #{@run_update}")
-
       duplicated_data_center_keys = select_duplicated_keys
 
-      duplicated_data_center_keys.each do |dc|
-        data_centers_with_validators_number = count_validators_and_gossip_nodes_in_data_center(dc)
-        sorted_data_centers = sort_data_centers_by_validators_number(data_centers_with_validators_number)
-
-        # Data center with most validators number
-        main_dc = set_main_data_center(sorted_data_centers)
-
-        sorted_data_centers.each do |entry|
-          log_processed_data_center(entry)
-
-          entry.data_center.data_center_hosts.each do |dch|
-            log_processed_data_center_host(dch)
-
-            main_dc_host = find_or_create_host(main_dc, dch)
+      duplicated_data_center_keys.each do |dc_key|
+        puts dc_key
+        sorted_data_centers = sort_data_centers_by_created_at(dc_key)
+        main_dc = sorted_data_centers.last
```

Comment:
I'd put main_dc here, so that user can check if correct DC was detected.
Or even better, all sorted_data_centers dates.